client/core/account.go

```diff
@@ -0,0 +1,132 @@
+package core
+
+import (
+	"encoding/hex"
+
+	"decred.org/dcrdex/client/db"
+	"github.com/decred/dcrd/dcrec/secp256k1/v3"
+)
+
+// AccountExport is used to retrieve account by host for export.
+func (c *Core) AccountExport(pw []byte, host string) (*Account, error) {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return nil, codedError(passwordErr, err)
+	}
+	_, err = addrHost(host)
+	if err != nil {
+		return nil, newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection and the dex.Asset for each asset.
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if !found {
+		return nil, newError(unknownDEXErr, "DEX: %s", host)
+	}
+
+	dc.acct.keyMtx.RLock()
+	accountId := dc.acct.id.String()
+	privKey := hex.EncodeToString(dc.acct.privKey.Serialize())
```

Comment:
This panics if account is not unlocked.  Either return with an error if the account is locked or unlock it for them

```go
	if err = dc.acct.unlock(crypter); err != nil {
		return nil, codedError(acctKeyErr, err)
	}
```